querier.go

```diff
@@ -420,14 +420,16 @@ func (s *populatedChunkSeries) Next() bool {
 				continue
 			}
 			if c.MinTime > s.maxt {
-				chks = chks[from+1 : i]
+				chks = chks[:i]
 				break
 			}
 			c.Chunk, s.err = s.chunks.Chunk(c.Ref)
 			if s.err != nil {
 				return false
 			}
 		}
+
+		chks = chks[from+1:]
```

Comment:
Can we kind of apply the same simple double for loop to get rid of tracking variables?